### ISSUE TITLE
TUNNELING_ACK is not detected

### ISSUE THREAD

**Issue by MaderM** [Supergiovane/node-red-contrib-knx-ultimate#114]
Hello,

i have a realy strange issue and after a few days i have to admit I'm not able to fix it on my own.

At first my setup:
I use a raspberry pi with nodred(v1.2.7) and knx ultimate(1.2.8).
On the KNX side a "Jung IPS 200 reg" is used as gateway (everythig has static IPs).
Further i use a "Jung 2316.16REGHE" as shutter actuator (and many more devices).

The setup runs without any problems. I can write and read on the KNX bus without failures.
But after a few switching operations on my shutter actuator (up and down), I get the following debug output:
"timed out waiting for TUNNELING_ACK"
After this i have to restart my flows to get it working again.

Here is a summary of what I've already found out:
-The Group monitor in ETS5 shows no abnormalities during the error
-In Wireshark I can see even in the error case "TunnelAck #A3:64 OK"
-The same error occurs also with knx easy (0.3.3) (this information may be usefule since easy and ultimate use the same code on many things)
-If I supress the ACK in the node settings i have the same issue
-Strange Part: If I remove the physical load from my actuator, I am NOT able to reproduce the error.

On the one hand, it looks like the actuator has a problem. On the other hand, I can neither see a false reading in the monitor nor in the Wireshark.
But I also wonder why I have to restart my flows just because an ACK has not arrived.

I'm running out of ideas. I've already tried to modify the code in FSM.js. But since I'm a C programmer, I feel like in hell right now.
I would be very happy if you take a look at my issue or have some tips.

Best regards
Mathias

PS: Since all other things work, I would like to express my compliments at this point. Keep it up!

**Comment by Supergiovane** [comment]
Hi Mader
It can be, that the ACK comes too late. It goes in timeout waiting for ack. 
The Jung actuaror may fault sending a properly formatted telegram and the ip interfaces takes longer to resume from the issue. 
For example, i have a Gira pushbutton that responds with a “write” telegram to “read” request. It should respond with “response” telegram instead. Not everytime, but often. 
Unfortunately, sometimes it happens that some devices behave abnormally. 
You can try to increase the delay betwren telegrams (there is a property in the config node), to see if it helps. 
You can also leave the “request ack” disabled (please restart nodered after this change). 

let me know!

**Comment by MaderM** [comment]
Hi,
thank you for the response.
Unfortunately, the proposed changes did not work.
I tried the whole thing again. This time I switched off the ACK and used a delay of 100ms between the messages.
This is the result:
It worked for 8 transfers. But after "timed out waiting for TUNNELING_ACK" I am not able to send more messages until i restart my flows.
ETS5 Group Monitor:
![ETS5GroupMonitor](https://user-images.githubusercontent.com/78514803/106936517-f90c2500-671c-11eb-85d5-6300f3d196fc.jpg)

NodeRed Log:
[log.txt](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/5927686/log.txt)

Wireshark Log:
![WiresharkLog](https://user-images.githubusercontent.com/78514803/106936555-06291400-671d-11eb-8f5d-c3e31f897e0b.jpg)

I wonder why the whole thing happens even though I suppress the ACK request.

Best regards
Mathias

**Comment by Supergiovane** [comment]
Do you have maybe a knx node from other developer installed?
If yes, please uninstall it.

**Comment by Supergiovane** [comment]
...then restart nodered.

**Comment by MaderM** [comment]
Hello,

I have already removed knx-easy before opening this issue.

Best regards
Mathias

**Comment by Supergiovane** [comment]
Tomorrow morning i’ll take a look deeply but without your actuator, it’s difficult. It must be something in the actuator itself. A misconfigured telegram. Maybe the magnetic field from the blind wires interfers with something inside the jung. Are you able to swap the jung with another identical jung actuator (maybe you have one other jung in your home)

**Comment by MaderM** [comment]
Hello,

I have now installed Wireshark on my Raspberry. It indicates that the IPS200 may have a problem. I think I'll get a new IP interface soon.
From other posts I read that a router is the better solution: "JUNG IPR 300 SREG KNX IP-Router"

I will respond as soon as there is something new.

Thank you and best regards
Mathias

**Comment by Supergiovane** [comment]
Hello Mathias
Having at least 2 knx/ip devices is good, because if one fails, you can swap ip in no time. 
You can even leave both operative and use the knx watchdog node to switch between the two, if one fails. 
The fact that you receive the Ack log message even if it isn’t enabled is strange and i’ll check that.

**Comment by Supergiovane** [comment]
Hello Mathias
are you able to unzip the attached file and replace your current FMS.js file, then restart nodered?
Try it and let me know if something changes.
[FSM.js.zip](https://github.com/Supergiovane/node-red-contrib-knx-ultimate/files/5930931/FSM.js.zip)

**Comment by MaderM** [comment]
Hello,

I finally found the cause. And I have to admit that I didn't expect that.

The problem is crosstalk between my bus line and the 230V load on my shutters. This leads to a malfunction of the IPS200. The IPS200 still reacts to all commands. But these are not forwarded to the KNX bus and are not answered with an ACK. If I restart my flows and a reconnect happens, the IPS200 seems to work correctly again.
Since I followed your tip and replaced my actuator, the routing of my bus line has also changed. When I put the old actuator back in, it worked. When I brought my bus line back to its original position, the error was there again.

So I really have to apologize to you for the inconvenience.
But the knowledge that crosstalk can be a problem even with a robust bus like KNX, may also help in the future.

Thank you for all the effort you have invested in my issue.
I think this issue can be closesd.

Best regards
Mathias
